### PR TITLE
Fix jpeg image flags and add jpeg codec tests

### PR DESCRIPTION
**Pull request by hughbe**
Fixes https://github.com/mono/libgdiplus/issues/260

### CODE REVIEW

File: src/jpegcodec.c [hunk 1]
```diff
@@ -344,19 +352,8 @@ gdip_load_jpeg_image_internal (struct jpeg_source_mgr *src, GpImage **image)
...
-		break;
-	case JCS_YCbCr:
-		result->active_bitmap->image_flags |= ImageFlagsColorSpaceYCBCR;
-		break;
-	case JCS_YCCK:
-		result->active_bitmap->image_flags |= ImageFlagsColorSpaceYCCK;
-		break;
-	case JCS_CMYK:
-		result->active_bitmap->image_flags |= ImageFlagsColorSpaceCMYK;
-		break;
```
Comment (filipnavara) [inline src/jpegcodec.c]: Removing this doesn't sound right. The failed test is weird though:

> Bitmap24bitFeatures (MonoTests.System.Drawing.Imaging.JpegCodecTest.Bitmap24bitFeatures)
>      Flags
>   Expected: 77960
>   But was:  77840

The expected value differs by combination of ImageFlagsColorSpaceYCCK and ImageFlagsColorSpaceCMYK.

File: src/jpegcodec.c [hunk 2]
```diff
@@ -302,10 +302,15 @@ gdip_load_jpeg_image_internal (struct jpeg_source_mgr *src, GpImage **image)
...
+	if (!result) {
+		status = OutOfMemory;
+		goto error;
+	}
+
 	result->type = ImageTypeBitmap;
 	result->active_bitmap->width = cinfo.image_width;
 	result->active_bitmap->height = cinfo.image_height;
-	result->active_bitmap->image_flags = ImageFlagsReadOnly | ImageFlagsHasRealPixelSize | ImageFlagsPartiallyScalable | ImageFlagsHasRealDPI;
+	result->active_bitmap->image_flags = ImageFlagsReadOnly | ImageFlagsHasRealPixelSize;
```
Comment (filipnavara) [inline src/jpegcodec.c]: Removing ImageFlagsPartiallyScalable breaks the following test:

> Bitmap8bbpIndexedGreyscaleFeatures (MonoTests.System.Drawing.Imaging.JpegCodecTest.Bitmap8bbpIndexedGreyscaleFeatures)
>      Flags
>   Expected: 77896
>   But was:  77888

File: src/jpegcodec.c [hunk 3]
```diff
@@ -318,6 +323,9 @@ gdip_load_jpeg_image_internal (struct jpeg_source_mgr *src, GpImage **image)
 		result->active_bitmap->dpi_vert = 0;
 	}
 
+	if (result->active_bitmap->dpi_horz && result->active_bitmap->dpi_vert)
+		result->active_bitmap->image_flags |= ImageFlagsHasRealDPI;
+
```
Comment (filipnavara) [inline src/jpegcodec.c]: 👍

File: src/jpegcodec.c (hunk 1 above)
Comment (filipnavara) [reply src/jpegcodec.c]: Eh, sorry, my bad... it's not hex :)

Comment (filipnavara) [reply src/jpegcodec.c]: Expected: ImageFlagsReadOnly | ImageFlagsHasRealPixelSize | ImageFlagsHasRealDPI | ImageFlagsColorSpaceYCBCR | ImageFlagsPartiallyScalable
But was: ImageFlagsReadOnly | ImageFlagsHasRealPixelSize | ImageFlagsHasRealDPI | ImageFlagsColorSpaceRGB

Comment (hughbe) [reply src/jpegcodec.c]: Also expected according to GDI+

**Pull request merged**